acpi/src/madt.rs

```diff
@@ -415,86 +415,86 @@ impl<'a> Iterator for MadtEntryIter<'a> {
 #[derive(Clone, Copy, Debug)]
 #[repr(C, packed)]
 pub struct EntryHeader {
-    entry_type: u8,
-    length: u8,
+    pub entry_type: u8,
+    pub length: u8,
 }
 
 #[derive(Clone, Copy, Debug)]
 #[repr(C, packed)]
 pub struct LocalApicEntry {
-    header: EntryHeader,
-    processor_id: u8,
-    apic_id: u8,
-    flags: u32,
+    pub header: EntryHeader,
+    pub processor_id: u8,
+    pub apic_id: u8,
+    pub flags: u32,
 }
 
 #[derive(Clone, Copy, Debug)]
 #[repr(C, packed)]
 pub struct IoApicEntry {
-    header: EntryHeader,
-    io_apic_id: u8,
-    _reserved: u8,
-    io_apic_address: u32,
-    global_system_interrupt_base: u32,
+    pub header: EntryHeader,
+    pub io_apic_id: u8,
+    pub _reserved: u8,
```

Comment:
Yep, I don't feel that reserved fields need to be exposed, and just clutter the documentation.

I don't foresee an instance where people will be constructing tables themselves from outside the library tbh, so that shouldn't be included in this PR.

Reply:
Makes sense. I updated the MR to not make the `_reserved` fields public.